### ISSUE TITLE
[Routing][DX] Deprecated bundle:controller:action Syntax Is Only Triggered on First Request

### ISSUE THREAD

**Issue by kiler129** [symfony/symfony#27387]
**Description**  
In my large project I still have routes defined as `bundle:controller:action`, which is deprecated since https://github.com/symfony/symfony/issues/25910. @Tobion did a great job on that PR, but the deprecation message has very poor DX.

PR for that issue mentions that deprecation message is displayed only during first request (which may be e.g. redirect) and is hard to spot. I believe the message should be displayed every time route is matched.

WDYT?

**Comment by Tobion** [comment]
> but the deprecation message has very poor DX

Can you elaborate?

> PR for that issue mentions that deprecation message is displayed only during first request

It happens when the container is built, not on the first request.

**Comment by stof** [comment]
> It happens when the container is built, not on the first request.

Actually, it happens during cache warmup, which makes things harder to spot.
Container-building deprecation are displayed by the profiler even on next requests since 3.4, but this is not the case for cache-warming deprecations.

**Comment by stof** [comment]
> I believe the message should be displayed every time route is matched.

the issue for that is that the cache warming is optimizing the router by resolving this notation to the class name during cache warmup. So when performing the matching itself in the request, the deprecation notation is not used.

**Comment by kiler129** [comment]
@Tobion As stof said it's very hard to spot these messages and they are very useful. Also our code contains a lot of internal deprecations, so it's hard to expect from developers to pay attention to a message which only appears once.

@stof I looked into the code and indeed it replaces `:` with `::` which makes checking after warming impossible. I also don't see a way to mark the route during the warming in any way...

My point is that any problem reported during compilation of container are persisted in profiler while problems with routes aren't. I showed this to few developers in the team and they were surprised such deprecation exists since they never seen it (in fact I myself found it not via profiler but while checking changelogs).
